plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -381,16 +381,15 @@ private void showRegularContent(@NotNull DBRProgressMonitor monitor) throws DBEx
         });
     }
 
-    private void showRestrictedContent(@NotNull DBDContent value) throws DBCException, IOException {
-        int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore().getInt(ResultSetPreferences.RS_EDIT_MAX_TEXT_SIZE);
+    private void showLimitedContent(@NotNull DBDContent value, int lengthInBytes) throws DBCException, IOException {
         DBDContentStorage contents = value.getContents(new VoidProgressMonitor());
         try (final InputStream stream = contents.getContentStream()) {
-            ByteArrayOutputStream buffer = new ByteArrayOutputStream((int) contents.getContentLength());
-            IOUtils.copyStream(stream, buffer);
-            byte[] displaingContentBytes = Arrays.copyOfRange(buffer.toByteArray(), 0, maxContentSize * 1000);
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream(lengthInBytes);
+            IOUtils.copyStream(stream, buffer, lengthInBytes);
```

Comment:
The third argument is the `bufferSize`, not the maximum amount of bytes to read.

Just create a byte array and use `stream.read(buffer)` to read it, it will return the length of bytes read which you can use to trim the output array if fewer bytes were read.

Reply:
Better to use pure readNBytes. 
`byte[] displaingContentBytes = stream.readNBytes(lengthInBytes);`